mkosi/gentoo.py

```diff
@@ -0,0 +1,508 @@
+            usrpath = root / "usr"
+
+            self.UNINSTALL_IGNORE = {
+                "/bin",
+                "/sbin",
+                "/lib",
+                "/lib64",
+            }
+
+            ignore = ["./var/run", "./var/lock", "./bin/awk"]
+            for dir in self.UNINSTALL_IGNORE:
+                ignore += [dir] + [f"./usr/{dir}"]
+
+            for tarinfo in members:
+                if tarinfo.name in ignore or re.match(devreg, tarinfo.name) or re.match(runreg, tarinfo.name):
+                    if "build-script" in ARG_DEBUG:
+                        print(f"{tarinfo.name} -> /dev/null")
+                    continue
+                elif (
+                    tarinfo.issym()
+                    and (os.path.dirname(tarinfo.name) == "./usr/bin")
+                    and (os.path.dirname(tarinfo.linkname) == "../../bin")
+                ):
+                    if "build-script" in ARG_DEBUG:
+                        print(f"{tarinfo.name} -> {tarinfo.linkname} -> /dev/null")
+                    continue
+                elif re.match(usrreg, tarinfo.name):
+                    usr.append(tarinfo)
+                elif re.match(binreg, tarinfo.name):
+                    bins.append(tarinfo)
+                else:
+                    print(f"{tarinfo.name} -> /dev/null")
+                    continue
+
+            try:
+                tfd.extractall(usrpath, bins, numeric_owner=True)
+                tfd.extractall(root, usr, numeric_owner=True)
+            except FileExistsError:
+                pass
+
+    def set_profile(self, args: CommandLineArguments, root: str) -> None:
+        # be careful not to overwrite user's profile, (via skeleton)
+        profile_path = root / PROFILE_PATH
+        if not os.path.islink(profile_path):
+            MkosiPrinter.print_step(f"{args.distribution} setting Profile")
+            os.makedirs(root / USER_CONFIG_PATH, exist_ok=True)
+            os.symlink(self.portage_cfg["PORTDIR"] / self.arch_profile, profile_path)
+
+    def set_default_repo(self, root: str) -> None:
+        os.makedirs(root / USER_CONFIG_PATH / "repos.conf", exist_ok=True)
+        with open(root / USER_CONFIG_PATH / "repos.conf" / "eselect-repo.conf", "w") as f:
+            f.write(
+                dedent(
+                    f"""\
+                    [gentoo]
+                    location = {self.portage_cfg["PORTDIR"]}
+                    sync-uri = https://anongit.gentoo.org/git/repo/gentoo.git
+                    sync-type = git
+                    sync-dept = 1
+                    """
+                )
+            )
+
+    def unmask_arch(self, root: str) -> None:
+        package_accept_keywords = root / USER_CONFIG_PATH / "package.accept_keywords"
+        os.makedirs(package_accept_keywords, exist_ok=True)
+        with open(package_accept_keywords / "mkosi", "a") as f:
+            # homed is still ~ARCH
+            f.write(
+                dedent(
```

Comment:
`.write_text()`, here and below…

Reply:
ditto

Reply:
done, i think, please let me know if missed something.